iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -273,6 +277,11 @@ def has_gpr_offsets(node: fx.Node) -> bool:
                         mapping=custom.mapping,
                         _write_dependency=custom._write_dependency,
                     ).add_to_graph(custom.graph)
+                elif isinstance(custom, SelfIndex):
+                    # TODO: Add support on how to handle strided reads.
+                    new_node = SelfIndex(
+                        custom.idx, custom.dtype, custom.elements_per_thread
```

Comment:
FYI: in line 202 above, `elements_per_thread` is taken as `custom.index[custom.idx].size` instead of the op argument. I don't fully understand what's going on here, so only pointing out the inconsistency.

Reply:
Good catch here, got some more clarification from Nicolas, fixed this up to be more consistent